src/BaseControllerV2.ts

```diff
@@ -258,16 +271,21 @@ export function getPersistentState<S extends Record<string, unknown>>(
   return deriveStateFromMetadata(state, metadata, 'persist');
 }
 
-function deriveStateFromMetadata<S extends Record<string, unknown>>(
+function deriveStateFromMetadata<
+  S extends Record<string, unknown>,
+  P extends keyof StatePropertyMetadata<S>
```

Comment:
I don't quite follow why this was made into a generic parameter. I'm going to take a look at this later to see whether this can be removed, to simplify the type signature a bit, and to remove the `as` below.

Reply:
Done in https://github.com/MetaMask/controllers/pull/483/commits/de4bf4627f0bad48b58d2d064cb4c7aa3c4fe9fc
it seems that everything worked fine without the generic parameter or the cast :shrug: